core/src/main/java/org/apache/iceberg/io/FileAccessFactory.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.io;
+
+import org.apache.iceberg.FileFormat;
+import org.apache.iceberg.data.Record;
+import org.apache.iceberg.deletes.PositionDelete;
+
+/**
+ * Interface that provides a unified abstraction for converting between data file formats and
+ * input/output data representations.
+ *
+ * <p>FileAccessFactory serves as a bridge between storage formats ({@link FileFormat}) and expected
+ * input/output data structures, optimizing performance through direct conversion without
+ * intermediate representations. File format implementations handle the low-level parsing details
+ * while the object model determines the in-memory representation used for the parsed data.
+ * Together, these provide a consistent API for consuming data files while optimizing for specific
+ * processing engines.
+ *
+ * <p>Iceberg provides these built-in object models:
```

Comment:
This list shouldn't be in core, even in a comment. It will just become stale and inaccurate.